libs/dotnet-sdk/State/FeatureBoardState.cs

```diff
@@ -1,60 +1,48 @@
+using System.Threading;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
 using FeatureBoard.DotnetSdk.Models;
-using System.Collections.Concurrent;
 
 namespace FeatureBoard.DotnetSdk.State;
 
 
-public class FeatureBoardState : IFeatureBoardState
+internal sealed class FeatureBoardState : IFeatureBoardState, IHostedService
 {
-  public DateTimeOffset? LastUpdated { get; private set; }
-  public string? ETag { get; private set; }
+  private readonly IServiceScopeFactory _scopeFactory;
   private readonly IFeatureBoardExternalState? _externalState;
-  private static readonly ConcurrentDictionary<string, FeatureConfiguration> _cache = new(StringComparer.OrdinalIgnoreCase);
 
-  public FeatureBoardState(IFeatureBoardExternalState? externalState = null)
+  private Dictionary<string, FeatureConfiguration> _cache = new(0);
+
+  public FeatureBoardState(IServiceScopeFactory scopeFactory, IFeatureBoardExternalState? externalState = null)
   {
+    _scopeFactory = scopeFactory;
     _externalState = externalState;
   }
 
-  public FeatureBoardStateSnapshot GetSnapshot() => new(_cache);
+  public FeatureBoardStateSnapshot GetSnapshot() => new FeatureBoardStateSnapshot(_cache);
```

Comment:
Because `Update` now creates a new dictionary on every call, there isn't really an issue with reference sharing anymore.

We could just return the _cache here unless the StateSnapshot has particular value, it just copies and provides an abstraction to Get. I guess it means no access to the dictionary mutation methods but I'm not super worried about that.

At a minimum, we could remove the copy inside the ctor as it should no longer be needed